packages/asset-buyer/src/quote_consumers/forwarder_swap_quote_consumer.ts

```diff
@@ -0,0 +1,228 @@
+        if (!this.isValidForwarderSwapQuote(quote)) {
+            throw new Error(AssetSwapQuoterError.InvalidForwarderSwapQuote);
+        }
+        const { ethAmount, feeRecipient, feePercentage: unFormattedFeePercentage } = _.merge(
+            {},
+            constants.DEFAULT_FORWARDER_SWAP_QUOTE_CONSUMER_FEE_OPTS,
+            opts,
+        );
+
+        assert.isValidSwapQuote('quote', quote);
+        assert.isNumber('feePercentage', unFormattedFeePercentage);
+        assert.isETHAddressHex('feeRecipient', feeRecipient);
+
+        if (!!ethAmount) {
+            assert.isBigNumber('ethAmount', ethAmount);
+        }
+
+        const swapQuoteWithFeeAdded = addAffiliateFeeToSwapQuote(quote, unFormattedFeePercentage);
+
+        const { orders, feeOrders, makerAssetBuyAmount: makerAssetFillAmount, worstCaseQuoteInfo } = swapQuoteWithFeeAdded;
+
+        const signatures = _.map(orders, o => o.signature);
+        const feeSignatures = _.map(orders, o => o.signature);
+
+        const feePercentage = utils.numberPercentageToEtherTokenAmountPercentage(unFormattedFeePercentage);
+
+
+        const params = {
+            orders,
+            makerAssetFillAmount,
+            signatures,
+            feeOrders,
+            feeSignatures,
+            feePercentage,
+            feeRecipient,
+        };
+
+        return {
+            params,
+            to: this._contractWrappers.forwarder.address,
+            value: ethAmount || quote.worstCaseQuoteInfo.totalTakerTokenAmount,
+        };
+    }
+
+    public executeSwapQuoteOrThrowAsync = async (quote: SwapQuote, opts: Partial<ForwarderSwapQuoteExecutionOpts>): Promise<string> => {
+        if (!this.isValidForwarderSwapQuote(quote)) {
+            throw new Error(AssetSwapQuoterError.InvalidForwarderSwapQuote);
+        }
+        const { ethAmount, takerAddress, feeRecipient, gasLimit, gasPrice, feePercentage } = _.merge(
+            {},
+            constants.DEFAULT_FORWARDER_SWAP_QUOTE_CONSUMER_FEE_OPTS,
+            opts,
+        );
+
+        assert.isValidSwapQuote('quote', quote);
+
+        if (ethAmount !== undefined) {
+            assert.isBigNumber('ethAmount', ethAmount);
+        }
+        if (takerAddress !== undefined) {
+            assert.isETHAddressHex('takerAddress', takerAddress);
+        }
+        assert.isETHAddressHex('feeRecipient', feeRecipient);
+        if (gasLimit !== undefined) {
+            assert.isNumber('gasLimit', gasLimit);
+        }
+        if (gasPrice !== undefined) {
+            assert.isBigNumber('gasPrice', gasPrice);
+        }
+
+        const swapQuoteWithFeeAdded = addAffiliateFeeToSwapQuote(quote, feePercentage);
+
+        const { orders, feeOrders, makerAssetBuyAmount, worstCaseQuoteInfo } = swapQuoteWithFeeAdded;
+
+        const finalTakerAddress = await this._getTakerAddressOrThrowAsync(opts);
+        try {
+            const txHash = await this._contractWrappers.forwarder.marketBuyOrdersWithEthAsync(
```

Comment:
Should this call be using `getSmartContractParamsOrThrow` above to derive the parameters?

Reply:
@dave4506 not sure I understand.

Reply:
That was a crappy explanation: the contract wrappers parameters do not map 1 to 1 to smart contract params. They do some additional logic to construct the smart contract params, redundant to do that work again before we call execute.